homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -130,15 +126,22 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 class Remote:
     """The Remote class. It stores the TV properties and the remote control connection itself."""
 
+    _control: RemoteControl | None = None
```

Comment:
```suggestion
    _control: RemoteControl
```
If you want to move type annotations to the class level, it mostly makes sense if you want to pretend it's always some type which it might not be at init time.

E.g. define `_control: RemoteControl` and in `__init__` set it to `self._control = None  # type: ignore[assignment]`. You wouldn't need all the asserts then.

Reply:
Yea I want to move it to injection

Reply:
As in, I dislike that we have to create the object in here, I want to create it in `async_setup_entry` and if we can't create it, raise `ConfigEntryNotReady`